telemetrylib/TCP.cpp

```diff
@@ -62,14 +68,38 @@ public slots:
         {
             QTcpSocket* sender = static_cast<QTcpSocket*>(QObject::sender());
             _sockets.removeOne(sender);
-            connected = false;
+            connection = false;
             emit connectionStatusChanged();
         }
     }
 private:
-    QTcpServer _server;     //the TCP socket
-    QList<QTcpSocket*> _sockets;    //The list of TCP socket
-    bool connected = false;     //This is used when getConnectionStatus() is called
+    QTcpServer _server;
+    QList<QTcpSocket*> _sockets;
+    std::atomic<bool> connection = false;
+    std::thread *t;
+    std::atomic<bool> isConnected=false;
+
+    /**
+     * creates a thread that pings the Chase car's static IP to check connection
+     */
+    void checkConnection() {
+        QTcpSocket sock;
+        while(!isConnected) {
+            sock.connectToHost("196.186.1.16", 22);
+            bool connected = sock.waitForConnected(500);
+            if (!connected && connection) {
+                sock.abort();
+                connection = false;
+                emit connectionStatusChanged();
+            } else if(connected && !connection) {
+                sock.close();
+                connection = true;
+                emit connectionStatusChanged();
+            }
+            usleep(50000);
+            sock.abort();
+        }
+    }
```

Comment:
I tested out Charles' changes with this as well as port 4005. Unfortunately, it doesn't look like it works. Could you please drop it from this PR for now?

Reply:
It does seem to be working with my testing on a multiple device network. Without this thread the class can't detect network a disconnect without tcp closing packet, with leads to data being lost during tcp timeout period.

Reply:
I added some printouts, but it's not connecting to the server on the engineering data distributor server. The connected/connection values are right after running connectToHost(). The TCP connection is good until the loop in the checking thread runs once.
```
send data current comm channel:  1
sending via TCP
send data current comm channel:  1
sending via TCP
in loop
connected (after con):  false
connection (after con):  true
Lost connection... aborting
comChanged!

 status:
0  : false 

1  : false 

send data current comm channel:  -1
adding to queue from telemetry:  1687690654833
```